### DISCUSSION TITLE
How do I get VSCode Vim Extension to Control VSCode Keybindings?

### DISCUSSION THREAD

**Discussion by Maxed0utt** [VSCodeVim/Vim#8172] [Q&A] [answered]
I am trying to get the Vim extension to run the VSCode shortcuts and I can't seem to figure out how to get it going.

for example:

In Vim i use <leader>t to open a terminal below. In VSCode this is Ctr+Shift+ or <leader>h and <leader>l to switch between splits is Ctr+1 and Ctr+2 in VSCode

I have tried remapping the Vim keys to the VSCode commands like this:

    "vim.normalModeKeyBindingsNonRecursive": [
    {
        "before": ["<leader>", "h"],
        "after": ["<C-1>"]
    },
]
but it doesn't seem to work.

Any help you can provide would be greatly appreciated!

**Comment by Acumane** [comment, marked as answer]
Not entirely sure what you're trying to bind, but try using `"commands"` instead of `"after"` to trigger events in VSCode. For example:
```
"vim.normalModeKeyBindings": [{
    "before": ["t"],
    "commands": ["workbench.action.togglePanel"],
    "silent": true
}]
```

**Reply by Maxed0utt** [reply]
THIS WORKED!!!! Thank you so much!

**Reply by Acumane** [reply]
Great! Are you able to mark this as the answer to close the thread?
